test/e2e_test/e2e_test.py

```diff
@@ -0,0 +1,153 @@
+from datetime import date, datetime
+import e2e_utils
+import json
+from jsonschema import validate
+from math import nan
+import numpy as np
+import os
+import pickle
+import torch
+from typing import List, Tuple
+import subprocess
+import sys
+sys.path.append('./..')
+from utils import BackendContext, SkylineSession
+
+
+class _ModelMetrics:
+    def __init__(self, name, batch_sizes=[], baseline_metrics=[], skyline_metrics=[]):
+        assert(len(batch_sizes) == len(baseline_metrics) == len(skyline_metrics))
+        self.name = name
+        self.batchSizes = batch_sizes
+        self.baselineThroughputs = [metric.samples_per_second for metric in baseline_metrics]
+        self.baselinePeakUsages = [metric.peak_usage_bytes for metric in baseline_metrics]
+        self.skylineThroughputs = [metric.samples_per_second for metric in skyline_metrics]
+        self.skylinePeakUsages = [metric.peak_usage_bytes for metric in skyline_metrics]
+        self.throughputPercentageDiffs = (np.abs(np.array(self.baselineThroughputs) - np.array(self.skylineThroughputs))/np.array(self.baselineThroughputs)).tolist()
+        self.peakUsagePercentageDiffs = (np.abs(np.array(self.baselinePeakUsages) - np.array(self.skylinePeakUsages))/np.array(self.baselinePeakUsages)).tolist()
+
+
+def _update_entry_point_batch_size(file_name, original_batch_size, target_batch_size):
+    if original_batch_size == target_batch_size:
+        pass
+    
+    with open(file_name, 'r') as file :
+        filedata = file.read()
+
+    # ensure that batch_size is a parameter
+    assert(filedata.find("batch_size=") != -1)
```

Comment:
This is going to bubble up to the user as an AssertionError. Should we add a try..except clause and throw something more user friendly?

Reply:
Good point I will update this